pkg/mimir/runtime_config_test.go

```diff
@@ -39,24 +40,24 @@ overrides:
 	runtimeCfg, err := loader.load(yamlFile)
 	require.NoError(t, err)
 
-	limits := validation.Limits{
-		IngestionRate:                       1500,
-		IngestionBurstSize:                  15000,
-		MaxGlobalSeriesPerUser:              15000,
-		MaxGlobalSeriesPerMetric:            7000,
-		RulerMaxRulesPerRuleGroup:           20,
-		RulerMaxRuleGroupsPerTenant:         20,
-		NotificationRateLimitPerIntegration: validation.NotificationRateLimitMap{},
-	}
+	expected := getDefaultLimits()
+	expected.IngestionRate = 1500
+	expected.IngestionBurstSize = 15000
+	expected.MaxGlobalSeriesPerUser = 15000
+	expected.MaxGlobalSeriesPerMetric = 7000
+	expected.RulerMaxRulesPerRuleGroup = 20
+	expected.RulerMaxRuleGroupsPerTenant = 20
 
 	loadedLimits := runtimeCfg.(*runtimeConfigValues).TenantLimits
 	require.Equal(t, 3, len(loadedLimits))
-	require.Equal(t, limits, *loadedLimits["1234"])
-	require.Equal(t, limits, *loadedLimits["1235"])
-	require.Equal(t, limits, *loadedLimits["1236"])
+	require.Equal(t, expected, *loadedLimits["1234"])
+	require.Equal(t, expected, *loadedLimits["1235"])
+	require.Equal(t, expected, *loadedLimits["1236"])
 }
 
 func TestRuntimeConfigLoader_ShouldLoadEmptyFile(t *testing.T) {
+	validation.SetDefaultLimitsForYAMLUnmarshalling(getDefaultLimits())
```

Comment:
you can add a TestMain() and call this once for the whole package as opposed to calling it in every test. Something along the lines of


```go
func TestMain(m *testing.M) {
	validation.SetDefaultLimitsForYAMLUnmarshalling(getDefaultLimits())
	m.Run()
}
```

same applies to pkg/util/validation/limits_test.go

Reply:
Yes, we can do it. The tricky thing is that we have some tests overriding the default limits, so we have to remember to revert back the default limits in the test cleanup: https://github.com/grafana/mimir/pull/7030/commits/3156fcf43de1b315bd325f8fed86510bd3abd5d9